From 114f347ab53e08e9c1b94056a29fa492c5a20a64 Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Wed, 20 Dec 2017 17:12:50 +0900
Subject: [PATCH 640/909] arm64: dts: r8a7795-es1: Override VCP4 device nodes
 for R-Car H3 ES1.x

commit d4d8f757cb25e9fa67aec8fb499090c25333adb0 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch overrides VCP4 device nodes for the R8A7795 ES1.x SoC.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi | 48 ++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi b/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
index bd4af636997f..72d0c0fa7000 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
@@ -46,6 +46,54 @@
 	/delete-node/ dma-controller@e6460000;
 	/delete-node/ dma-controller@e6470000;
 
+	vcp4@fe8f0000 {
+		compatible = "renesas,vcp4-vcpl4";
+		reg = <0 0xfe8f0000 0 0x200>, <0 0xfe8f0200 0 0x200>;
+		interrupts = <GIC_SPI 258 IRQ_TYPE_LEVEL_HIGH>,
+			     <GIC_SPI 259 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cpg CPG_MOD 129>;
+		power-domains = <&sysc R8A7795_PD_A2VC0>;
+		resets = <&cpg 129>;
+		renesas,#ch = <1>;
+		renesas,#fcp_ch = <1>;
+	};
+
+	vcp4@fe900000 {
+		renesas,#ch = <2>;
+	};
+
+	vcp4@fe8d0000 {
+		renesas,#ch = <3>;
+		renesas,#fcp_ch = <2>;
+	};
+
+	fcpci0: vcp4@fe8df000 {
+		compatible = "renesas,vcp4-fcpci0";
+		reg = <0 0xfe8df000 0 0x200>;
+		clocks = <&cpg CPG_MOD 617>;
+		power-domains = <&sysc R8A7795_PD_A3VC>;
+		resets = <&cpg 617>;
+		renesas,#ch = <2>;
+	};
+
+	fcpci1: vcp4@fe8ff000 {
+		compatible = "renesas,vcp4-fcpci1";
+		reg = <0 0xfe8ff000 0 0x200>;
+		clocks = <&cpg CPG_MOD 616>;
+		power-domains = <&sysc R8A7795_PD_A3VC>;
+		resets = <&cpg 616>;
+		renesas,#ch = <1>;
+	};
+
+	stb: vcp4@fe8a0000 {
+		compatible = "renesas,vcp4-stb";
+		reg = <0 0xfe8a0000 0 0x200>;
+		interrupts = <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cpg CPG_MOD 104>;
+		power-domains = <&sysc R8A7795_PD_A3VC>;
+		resets = <&cpg 104>;
+	};
+
 	vspm@fe920000 {
 		renesas,#rpf_clut = <4>;
 		/delete-property/ renesas,#start_reservation;
-- 
2.17.1

